File: Arrays and Hashing/242-Valid-Anagram/242.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) {
            return false;
        }
        HashMap<Character, Integer> s_map = new HashMap<>();
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            s_map.put(c, s_map.getOrDefault(c, 0) + 1);
        }
        for (int i = 0; i < t.length(); i++) {
            char c = t.charAt(i);
            if (!s_map.containsKey(c)) {
                return false;
            }
            s_map.put(c, s_map.get(c) - 1);
            if (s_map.get(c) == 0) {
                s_map.remove(c);
            }
        }
        return s_map.isEmpty();
    }
}

File: Arrays and Hashing/242-Valid-Anagram/242.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.size() != t.size()){
            return false;
        }
        std::unordered_map<char, int> s_map;
        for(int i = 0; i < s.size(); i++){
            if(s_map.contains(s[i])){
                s_map[s[i]]++;
            }
            else{
                s_map[s[i]] = 1;
            }
        }
        for(int i = 0; i < t.size(); i++){
            if(s_map.contains(t[i])){
                if(s_map[t[i]] == 0){
                    return false;
                }
                else{
                    s_map[t[i]]--;
                }
            }
            else{
                return false;
            }
        }
        return true;
    }
};